### PR TITLE
Update build tools, target android and various dependencies

### PR DESCRIPTION
**Pull request by jaredsburrows**
@JakeWharton

### CODE REVIEW

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -15,12 +15,12 @@ ext {
       pollexor           : 'com.squareup:pollexor:2.0.4',
       supportV4          : "com.android.support:support-v4:$supportLibrariesVersion",
       supportAnnotations : "com.android.support:support-annotations:$supportLibrariesVersion",
-      junit              : 'junit:junit:4.10',
+      junit              : 'junit:junit:4.12',
       fest               : 'org.easytesting:fest-assert-core:2.0M10',
       festAndroid        : 'com.squareup:fest-android:1.0.6',
       robolectric        : 'org.robolectric:robolectric:3.1',
```
Comment (piotrek1543) [inline gradle/dependencies.gradle]: what about `robolectric` and `okhtttp` libs upgrade?

Comment (jaredsburrows) [reply gradle/dependencies.gradle]: They break the unit tests. I didn't feel like making this too big a PR.

**Comment by jaredsburrows** [comment]
It is weird that I got no notifications for this failing PR. It is missing the licenses.

**Comment by jaredsburrows** [comment]
@piotrek1543 @JakeWharton Not it is failing because of lint. We need to merge https://github.com/square/picasso/pull/1663 first.

**Comment by JakeWharton** [comment]
Please rebase now that the other PR is merged.

**Comment by jaredsburrows** [comment]
@JakeWharton Thanks. Rebased and build restarted.

**Pull request merged**